namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -170,26 +171,75 @@ class HttpControlService(storage: DtabStore, delegate: Ns => NameInterpreter, na
       resp
   }
 
-  private[this] def handleList(): Future[Response] =
-    storage.list().map { namespaces =>
-      val rsp = Response()
-      rsp.contentType = MediaType.Json
-      rsp.content = Json.write(namespaces)
-      rsp
+  private[this] def streamingResp[T](
+    values: Event[Try[T]],
+    contentType: Option[String] = None
+  )(render: T => Buf): Future[Response] = {
+    val resp = Response()
+    for (ct <- contentType) resp.contentType = ct
+    resp.setChunked(true)
+    val writer = resp.writer
+    // closable is a handle to the values observation so that we can close the observation when the
+    // streaming connection is terminated
+    @volatile var closable: Closable = Closable.nop
+    // calls to writer.write must be flatMapped together to ensure proper ordering and backpressure
+    // writeFuture is an accumulator of those flatMapped Futures
+    @volatile var writeFuture: Future[Unit] = Future.Unit
+    closable = values.respond {
+      case Return(t) =>
+        writeFuture = writeFuture.before {
+          val buf = render(t)
+          if (buf == Buf.Empty)
+            Future.Unit
+          else
+            writer.write(buf).onFailure { _ => closable.close() }
+        }
+      case Throw(e) =>
+        val _ = writer.write(Buf.Utf8(e.getMessage).concat(newline)).before(writer.close())
```

Comment:
Should we make render implement `Try[T] => Buf`?  For json streams, in particular, it seems sketchy to return a raw string in a sequence of json objects.  It would be nice if it could be encoded as json...